src/contract/dynamiccontract.h

```diff
@@ -52,6 +71,47 @@ class DynamicContract : public BaseContract {
     inline void registerVariableUse(SafeBase& variable) { usedVariables.emplace_back(variable); }
 
   protected:
+    /**
+     * Helper function for registering a payable/non-payable function.
+     * @tparam R Return type of the function.
+     * @tparam T Class type.
+     * @tparam MemFunc Member function type.
+     */
+    template <typename R, typename T>
+    struct RegisterHelper {
+        template <typename MemFunc>
+        /**
+        * Create a ReturnType object by calling the member function.
+        * @param instance Pointer to the instance of the class.
+        * @param memFunc Pointer to the member function.
+        * @return A ReturnType object.
+        */
+        static ReturnType createReturnType(T* instance, MemFunc memFunc) {
+            return ReturnType((instance->*memFunc)());
+        }
+    };
+
+    /**
+    * Helper function for registering a void payable/non-payable function.
+    * @tparam T Class type.
+    * @tparam MemFunc Member function type.
+    */
+    template <typename T>
+    struct RegisterHelper<void, T> {
+        template <typename MemFunc>
+        /**
+        * Create a ReturnType object by calling the member function.
+        * @param instance Pointer to the instance of the class.
+        * @param memFunc Pointer to the member function.
+        * @return A ReturnType object (default constructed).
+        */
+        static ReturnType createReturnType(T* instance, MemFunc memFunc) {
+            (instance->*memFunc)(); // Call the function even though it's void
+            // Handle void case here: return a default ReturnType, or throw an exception, etc.
```

Comment:
If this is an important thing to look at later on, I would suggest tagging it as a TODO so we can track it.

Reply:
Done!